src/cli/utils.js

```diff
@@ -38,46 +38,33 @@ function getAPICtl (apiAddr) {
   return APIctl(apiAddr)
 }
 
-exports.getIPFS = (argv, callback) => {
+exports.getNodeOrAPI = (argv, stateOptions = {forceInitialized: true}) => {
+  log('get node or api async')
   if (argv.api || isDaemonOn()) {
-    return callback(null, getAPICtl(argv.api), (cb) => cb())
+    return Promise.resolve(getAPICtl(argv.api))
   }
 
-  // Required inline to reduce startup time
   const IPFS = require('../core')
-  const node = new IPFS({
+  return IPFS.createNodePromise({
     repo: exports.getRepoPath(),
     init: false,
     start: false,
```

Comment:
Also here, does `createNodePromise` resolve when `start` is `false`?

Reply:
yes, ready is always emitted if an error doesn't happen
https://github.com/ipfs/js-ipfs/blob/feat/better-errors-cli/src/core/boot.js#L53